### PR TITLE
[CMS-1179] Release notes for upcoming drupal-composer-managed release supporting D10

### PR DESCRIPTION
**Pull request by namespacebrian**
## Summary

Adds a new start-state document for Drupal, alongside the existing one for WordPress, to house changelog information for us to link to from dashboard updates, and creates a placeholder for our upcoming drupal-composer-managed update which provides some D10 compatibility updates.

**[Doc Page Title](https://docs.pantheon.io/doc-title)** - <Enter a one sentence summation of the pertinent changes (including not-yet-completed work) provided by this PR.>

## Effect

Adds a new start-state document for Drupal, alongside the existing one for WordPress, to house changelog information for us to link to from dashboard updates, and creates a placeholder for our upcoming drupal-composer-managed update which provides some D10 compatibility updates.

The following changes are already committed:

* creation of source/content/start-states/drupal-composer-managed.md
* placeholder for upcoming release notes

## Remaining Work and Prerequisites

The following changes still need to be completed:

- [x] Add actual language for the upcoming release

### Dependencies and Timing

**Release**:
- [x] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/old-path/` => `/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

File: source/content/start-states/drupal-composer-managed.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- The enable-patching flag is added to the site's top-level composer.json file. Patches from dependencies may not be applied if this option is missing, so we are including it to avoid confusion. The most recent version of `cweagans/composer-patches` still requires this flag, but it is slated to be removed in a future release. If you do not want this flag enabled in your project’s composer.json file, you may remove it, and it will not be replaced.
+- The phpstan/extension-enstaller Composer plugin is enabled, if it isn't already, to reduce friction with future upgrades to Drupal 10, which includes this extension.
+
+These changes have minimal impact on existing sites today, and are primarily intended to keep the Drupal (Composer Managed) upstream in a state where it is easy for Pantheon to maintain the new Drupal 10 "start state" upstream (see [https://github.com/pantheon-upstreams/drupal-10-composer-managed](https://github.com/pantheon-upstreams/drupal-10-composer-managed)).
```
Comment (greg-1-anderson) [inline source/content/start-states/drupal-composer-managed.md:31]: Maybe the human-readable text should just be `pantheon-upstreams/drupal-10-composer-managed`. Up to docs, follow standards for linking to github repo (if you have defined any).

File: source/content/start-states/drupal-composer-managed.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+## Previous Releases
+
+### 2023-THE-PAST
+
+<a name="20230117" class="release-update"></a>Do we want to actually put any historical release information in this document? 🤔
```
Comment (greg-1-anderson) [inline source/content/start-states/drupal-composer-managed.md:38]: I'd just list two:

- 2022 Dec 13: Update to PHP 8.1
- 2022 Mar 24: Allow sites to upgrade to Drush 12 once it becomes available

**Review by greg-1-anderson — APPROVED** [review]

File: source/content/start-states/drupal-composer-managed.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+audience: [agency, business, development]
+product: [custom-upstreams]
+integration: [--]
+---
+
+Composer-managed Drupal sites!
+
+## Latest Release
+
+### 2023-02-22
```
Comment (greg-1-anderson) [inline source/content/start-states/drupal-composer-managed.md:22]: Do we want this to be 2023-02-23 now?

**Comment by whitneymeredith** [comment]
[Preview](https://pr-8164-documentation.appa.pantheon.site/start-states/drupal-composer-managed)

**Comment by greg-1-anderson** [comment]
The most recent commit is the same as the preview above, except it changes the date to the 23rd (tomorrow). The preview should catch up shortly.

**Review by whitneymeredith — APPROVED** [review]
LGTM

**Pull request merged**